src/gpu/nvidia/cudnn_matmul_executor.hpp

```diff
@@ -392,12 +392,12 @@ struct cudnn_matmul_lt_exec_t final : public cudnn_matmul_lt_base_exec_t {
                             memory_tracking::names::key_matmul_dst_in_acc_dt)
                     : xpu::sycl::interop_memory_arg_t<
                             ::sycl::access::mode::read_write>();
-            auto arg_block_a_scratch = params->source_size_ != 0
+            auto arg_block_a_scratch = params->weight_size_ != 0
```

Comment:
(random spot) the cudnn_matmul commits seem to be unrelated to generic inner product.
Could you commit those fixes in a separate commit (same PR is fine), this way if one or the other needs to be reverted, we don't revert everything.

Reply:
Sure, this is done